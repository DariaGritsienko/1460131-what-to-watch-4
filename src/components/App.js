import React from 'react';
import Main from './Main';
import About from './About';
import PropTypes from 'prop-types';
import {BrowserRouter, Route, Switch} from 'react-router-dom';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.onFilmsTitleClick = this.onFilmsTitleClick.bind(this);
    this.state = {
      onFilmsTitleClick: ``,
      aboutFilm: [],
    };
  }

  onFilmsTitleClick(isClick) {
    this.setState({onFilmsTitleClick: isClick});
  }

  _renderAbout(filmElement) {
    const {store} = this.props;
    const arr = [];
    if (!filmElement) {
      return null;
    }
    store.getState().films.map((film) => {
      arr.push(filmElement.genre === film.genre && filmElement.title !== film.title ? film : null);
    });
    return (
      <About
        filmData={filmElement}
        moreFilms={arr.filter((item) => !!item)}
        onFilmsTitleClick={(e) => {
          this.onFilmsTitleClick(e);
        }}
      />
    );
  }

  render() {
    const {store} = this.props;
    if (!store) {
      return null;
    }
    const filmData = store.getState().films.find((film) => {
      if (film.title === this.state.onFilmsTitleClick) {
        return true;
      }
      return false;
    });

    return (
      <BrowserRouter>
        <Switch>
          <Route exact path='/'>
            <Main
              filmsData={store.getState().films}
              store={store}
              onFilmsTitleClick={(e) => {
                this.onFilmsTitleClick(e);
              }}
            />
          </Route>
          <Route path='/about'>
            {this._renderAbout(filmData)}
          </Route>
        </Switch>
      </BrowserRouter>
    );
  }
}

App.propTypes = {
  store: PropTypes.object
};
