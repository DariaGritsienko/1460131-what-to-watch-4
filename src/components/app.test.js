import React from "react";
import renderer from "react-test-renderer";
import App from './App';

const filmsData = {
  'genres': [`Comedy`],
  'years': [`1234`],
  'titles': [`Some film1`, `Some film2`, `Some film3`]
};

describe(`Foo`, () => {
  it(`<App /> should render main page in app`, () => {
    const tree = renderer.create(
        <App
          filmsData={filmsData}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
