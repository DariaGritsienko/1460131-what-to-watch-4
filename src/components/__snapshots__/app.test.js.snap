// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foo <App /> should render main page in app 1`] = `null`;
